vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -469,3 +611,51 @@ pub(crate) fn admin_cmd(opcode: spec::AdminOpcode) -> spec::Command {
         ..FromZeroes::new_zeroed()
     }
 }
+
+#[repr(C)]
+#[derive(Protobuf, Clone, Debug, AsBytes, FromBytes, FromZeroes)]
+#[mesh(package = "underhill")]
+pub struct PendingCommandSavedState {
+    #[mesh(1)]
+    pub command: [u8; 64],
+    #[mesh(2)]
+    pub cid: u16,
+}
+
+impl From<&[u8]> for PendingCommandSavedState {
```

Comment:
Why are we doing this instead of using protobuf?

Reply:
This is my implementation to re-create PendingCommand from byte array. I did not find a better way, open to suggestions.

Reply:
Why do we need to re-create it from a byte array?

Reply:
Changed to protobuf as part of CID-high-bits integration and now spec::Command has dependency on Protobuf. If that's not ok then I'll revert.